Assets/Scripts/Model/Weapon.cs

```diff
@@ -98,7 +105,7 @@ public CanReceiveDamage getAvailableTarget()
             var target = targets[0];
 
             // Check if target is already dead
-            if (target.Equals(null))
+			if (target.Equals(null) || target == null)
```

Comment:
I'll try to remove it and see if it is necessary, because I think the problem wasn't really in this line.